L/LightGBM/build_tarballs.jl

```diff
@@ -17,27 +22,46 @@ cd $WORKSPACE/srcdir/LightGBM
 git submodule update --init --depth=1
 git submodule update --checkout --depth=1
 
-
-if [[ $target == *"apple-darwin"* ]]; then
-  cmake_extra_args="-DAPPLE=1 -DAPPLE_OUTPUT_DYLIB=1"
-fi
+for p in $WORKSPACE/srcdir/patches/*.patch; do
+  atomic_patch -p1 "${p}"
+done
 
 FLAGS=()
+cmake_extra_args=""
+
+if [[ "${bb_full_target}" == *"apple-darwin"* ]]; then
+  cmake_extra_args+="-DAPPLE=1 -DAPPLE_OUTPUT_DYLIB=1 "
+fi
 
-if [[ "${target}" == *-mingw* ]]; then
-  # Parches windows
-  for p in $WORKSPACE/srcdir/patches/windows/*.patch; do
-    atomic_patch -p1 "${p}"
-  done
-  cmake_extra_args="-DWIN32=1 -DMINGW=1"
+if [[ "${bb_full_target}" == *-mingw* ]]; then
+  cmake_extra_args+="-DWIN32=1 -DMINGW=1 "
   FLAGS+=(LDFLAGS="-no-undefined")
 fi
 
-cmake \
+if [[ "${bb_full_target}" == *-linux* ]]; then
+  cmake_extra_args+="-DUSE_GPU=1 "
+fi
+
+if  [[ ("${bb_full_target}" == *-cuda*) && ("${bb_full_target}" != *-cuda+none*) ]]; then
+  # nvcc writes to /tmp, which is a small tmpfs in our sandbox.
+  # make it use the workspace instead
+  export TMPDIR=${WORKSPACE}/tmpdir
+  mkdir ${TMPDIR}
+  export CUDA_PATH=${WORKSPACE}/destdir/cuda
+  export PATH=$PATH:$CUDA_PATH/bin/
+
+  ln -s $CUDA_PATH/lib $CUDA_PATH/lib64
```

Comment:
I think this leaves a broken symlink in the final tarballs?

Reply:
I found that the issue was in the enable_language(CUDA) call, it was not using the library path provided, so I'm forcing it now with CMAKE_CUDA_FLAGS, is this a good approach?